src/datastore/postgres-store.ts

```diff
@@ -2159,6 +2148,94 @@ export class PgDataStore
+              ORDER BY canonical DESC, microblock_canonical DESC, block_height DESC
+              LIMIT 1`,
+              [hexToBuffer(attachment.txId), hexToBuffer(attachment.indexBlockHash)]
+            );
+            let isCanonical = true;
+            let txIndex = -1;
+            const blockData: DataStoreSubdomainBlockData = {
+              index_block_hash: '',
+              parent_index_block_hash: '',
+              microblock_hash: '',
+              microblock_sequence: I32_MAX,
+              microblock_canonical: true,
+            };
+            if (dbTx.rowCount > 0) {
+              const parsedDbTx = this.parseTxQueryResult(dbTx.rows[0]);
+              isCanonical = parsedDbTx.canonical;
+              txIndex = parsedDbTx.tx_index;
+              blockData.index_block_hash = parsedDbTx.index_block_hash;
+              blockData.parent_index_block_hash = parsedDbTx.parent_index_block_hash;
+              blockData.microblock_hash = parsedDbTx.microblock_hash;
+              blockData.microblock_sequence = parsedDbTx.microblock_sequence;
+              blockData.microblock_canonical = parsedDbTx.microblock_canonical;
+            } else {
+              logger.warn(
+                `Could not find transaction ${attachment.txId} associated with attachment`
+              );
+            }
+            const subdomains: DbBnsSubdomain[] = [];
+            for (let i = 0; i < zoneFileTxt.length; i++) {
+              const zoneFile = zoneFileTxt[i];
+              const parsedTxt = parseZoneFileTxt(zoneFile.txt);
+              if (parsedTxt.owner === '') continue; //if txt has no owner , skip it
+              const subdomain: DbBnsSubdomain = {
+                name: attachment.name.concat('.', attachment.namespace),
+                namespace_id: attachment.namespace,
+                fully_qualified_subdomain: zoneFile.name.concat(
+                  '.',
+                  attachment.name,
+                  '.',
+                  attachment.namespace
+                ),
+                owner: parsedTxt.owner,
+                zonefile_hash: parsedTxt.zoneFileHash,
+                zonefile: parsedTxt.zoneFile,
+                tx_id: attachment.txId,
+                tx_index: txIndex,
+                canonical: isCanonical,
+                parent_zonefile_hash: attachment.zonefileHash.slice(2),
+                parent_zonefile_index: 0,
+                block_height: attachment.blockHeight,
+                zonefile_offset: 1,
+                resolver: zoneFileContents.uri ? parseResolver(zoneFileContents.uri) : '',
+              };
+              subdomains.push(subdomain);
+            }
+            subdomainData.push({ blockData, subdomains, attachment });
+          }
+        }
+        await this.updateBatchSubdomainsAndZonefiles(client, subdomainData);
+        // Update main zonefile as well.
+        await this.updateBatchSubdomainsAndZonefiles(client, [{ attachment }]);
```

Comment:
I'm a bit confused at what's going on here. It looks like the `updateBatchSubdomainsAndZonefiles` function is a bit overloaded and maybe performing too many different things?

Reply:
Good point, I was carrying the pattern from the old code but I can definitely split it up